src/com/rcv/CVRReader.java

```diff
@@ -48,26 +48,20 @@ public boolean parseCVRFile(String excelFilePath,
       RCVLogger.log("invalid RCV format: not enough rows:%d", contestSheet.getLastRowNum());
       return false;
     }
-    
+
+    // extract file name -- this along with ballot index will be used to generate ballot IDs
+    File inFile = new File(excelFilePath);
+    String cvrFileName = inFile.getName();
+    int ballotIndex = 1;
     // Iterate through all rows and create a CastVoteRecord for each row
     while (iterator.hasNext()) {
       org.apache.poi.ss.usermodel.Row castVoteRecord = iterator.next();
 
-      // create object for this row
-      ArrayList<ContestRanking> ballot = new ArrayList<ContestRanking>();
+      // TODO: determine how ballot IDs will be handled for different ballot styles
+      String ballotID =  String.format("%s(%d)",cvrFileName,ballotIndex++);
 
-      // parse ID
```

Comment:
I think we should keep this logic. When the ballot does contain IDs, let's use them in addition to our own IDs. We could just have two separate vars in the CVR class.

Reply:
Going to leave as is until we hear otherwise from product.  It's easy to put back if we need to.